packages/sns/src/errors/governance.errors.ts

```diff
@@ -0,0 +1 @@
+export class SnsGovernanceError extends Error {}
```

Comment:
would you have spontaneously another idea than using class for errors?

context: we should be precocious about this I think when we use it so I would welcome any cool idea

right now we lazy load sns-js  in a nice way in nns-dapp

but, on the contrary, I've got the feeling that loading errors and utils of nns-js in nns-dapp makes and would make hard(er) to lazy load.

rollup in nns-dapp currently does no chunk the libs so as soon as an error or utils is fetched boom all the lib is loaded it seems

Reply:
Good point. Yet, I don't think that the design decisions should take that into account. We might end up adding a complexity for all developers when they might not want.

If nns-dapp can't lazy load something, it should be fixed there, not from the library.

We could have also used feature flags to render or not a component that then brings all a library.

Or as you say, change or improve our chunking process.

That said, I don't have a problem to change errors from classes to something else. But I can't think of anything better now.

Reply:
That's a good point. I agree and disagree (a bit) in the sense that if we can avoid the all community to have to load kb of javascript, it would make my day 😉. But yes there are other way to prevent this (like doc and tutorial).

Let's go forward. When you gonna use it in nns-dapp, can you have a look that everything is still alright? (sns should not be loaded on sign-in screen). Thanks.